backends/gdx-backend-robovm/src/com/badlogic/gdx/backends/iosrobovm/objectal/OALIOSAudioDevice.java

```diff
@@ -0,0 +1,125 @@
+
+package com.badlogic.gdx.backends.iosrobovm.objectal;
+
+import com.badlogic.gdx.audio.AudioDevice;
+import org.robovm.rt.bro.Struct;
+import org.robovm.rt.bro.ptr.ShortPtr;
+import org.robovm.rt.bro.ptr.VoidPtr;
+
+import java.nio.ShortBuffer;
+import java.util.ArrayList;
+
+/** @author Jile Gao
+ * @author Bernstanio */
+class OALIOSAudioDevice implements AudioDevice {
+	private ALSource alSource;
+	private ArrayList<ALBuffer> alBuffers = new ArrayList<>();
+	private ArrayList<ALBuffer> alBuffersFree = new ArrayList<>();
+	private final int samplingRate;
+	private final boolean isMono;
+	private final int format;
+	private final ShortBuffer tmpBuffer;
+	private final int minSize;
+	private final int latency;
+
+	OALIOSAudioDevice (int samplingRate, boolean isMono, int minSize, int bufferCount) {
+		this.samplingRate = samplingRate;
+		this.isMono = isMono;
+		this.format = isMono ? 0x1101 : 0x1103;
+		this.minSize = minSize;
+
+		tmpBuffer = ShortBuffer.allocate(minSize);
+		latency = minSize / (isMono ? 1 : 2) / bufferCount;
+		alSource = new ALSource();
+		for (int i = 0; i < bufferCount; i++) {
+			ALBuffer buffer = new ALBuffer().initWithNameDataSizeFormatFrequency("test", Struct.allocate(VoidPtr.class, 1), 2,
+				format, samplingRate);
+			alBuffersFree.add(buffer);
+		}
+	}
+
+	@Override
+	public boolean isMono () {
+		return isMono;
+	}
+
+	@Override
+	public void writeSamples (short[] samples, int offset, int numSamples) {
+		if (numSamples < 0) throw new IllegalArgumentException("numSamples cannot be < 0.");
+		ShortPtr shortPtr;
+		if (numSamples + tmpBuffer.position() >= minSize) {
+			shortPtr = Struct.allocate(ShortPtr.class, numSamples + tmpBuffer.position());
+
+			shortPtr.set(tmpBuffer.array(), 0, tmpBuffer.position());
+			shortPtr.set(samples, offset + tmpBuffer.position(), numSamples);
```

Comment:
This is not correct.
`tmpBuffer.position()` is a destination offset and `offset` is a source offset.
I _think_ you need to do something like:
`shortPtr.next(tmpBuffer.position()).set(samples, offset, numSamples);`